packages/server/lib/controllers/v1/connectUISettings/putConnectUISettings.ts

```diff
@@ -38,9 +38,20 @@ export const putConnectUISettings = asyncWrapper<PutConnectUISettings>(async (re
         return;
     }
 
-    const { environment } = res.locals;
+    const { environment, plan } = res.locals;
     const body: PutConnectUISettings['Body'] = val.data;
 
+    const newSettings: ConnectUISettings = body;
+
+    // Override settings to defaults if the plan does not have the feature
+    if (flagHasPlan && !plan?.can_customize_connect_ui_theme) {
+        newSettings.theme = connectUISettingsService.defaultConnectUISettings.theme;
+    }
+
+    if (flagHasPlan && !plan?.can_disable_connect_ui_watermark) {
+        newSettings.showWatermark = connectUISettingsService.defaultConnectUISettings.showWatermark;
+    }
```

Comment:
would it make more sense to return an error (405 maybe) instead of accepting the request when plan doesn't allow customization